packages/react-tooltip/src/components/Tooltip/Tooltip.types.ts

```diff
@@ -101,6 +136,32 @@ export type TooltipTriggerProps = Pick<
   'onPointerEnter' | 'onPointerLeave' | 'onFocus' | 'onBlur' | 'aria-describedby' | 'aria-labelledby' | 'aria-label'
 >;
 
+/**
+ * Data for the Tooltip's onBeforeShow event.
+ */
+export interface OnBeforeShowTooltipData {
+  /**
+   * The element that the tooltip will point to. By default, this is the element that triggered the tooltip
+   * (event.currentTarget). It can be changed by the onBeforeShow handler to have the tooltip point to another element.
+   */
+  target: HTMLElement;
+
+  /**
+   * Set preventShow to true to cancel showing the tooltip.
+   */
+  preventShow?: boolean;
```

Comment:
This kind of flag makes the API surface for our components far more complicated than a user passing in `true/false` values to show the tooltip which they can more easily control from their application.